src/core/lib/surface/completion_queue.c

```diff
@@ -961,7 +960,7 @@ static void cq_finish_shutdown_next(grpc_exec_ctx *exec_ctx,
   cq_next_data *cqd = (cq_next_data *)DATA_FROM_CQ(cq);
 
   GPR_ASSERT(cqd->shutdown_called);
-  GPR_ASSERT(gpr_atm_no_barrier_load(&cqd->pending_events) == 0);
```

Comment:
As a matter of convention and sanity, this function should probably be called cq_finish_shutdown_next_locked since it has to be called under a lock.

Reply:
Good point - we haven't been very consistent about these in completion queue and polling engine code (for example, `pollset_work` requires a lock on pollset) . Perhaps we can do a separate clean-up PR to address this ?

Reply:
Yes, separate cleanup PR is fine